core/src/smartcontracts/isi/asset.rs

```diff
@@ -335,7 +334,89 @@ pub mod isi {
                     .ok_or(MathError::Overflow)?;
                 wsv.metrics.tx_amounts.observe(f64::from(*quantity));
 
-                Ok(AssetEvent::Added(destination_asset_id.clone()))
+                Ok(AssetEvent::Added(self.destination_id.clone()))
+            })
+        }
+    }
+
+    impl<W: WorldTrait> Execute<W> for Transfer<Asset, u128, Asset> {
```

Comment:
Transfer functions here look like it can be implemented with a generic function

Reply:
So can be minting/burning. Do you prefer a marker trait for asset values, or explicit trait bounds on all objects?

Reply:
Created #2204 .

Reply:
Hmm, let's an author of #2204 changes will decide the best apprroach